dali/operators/reader/numpy_reader_gpu_op.cc

```diff
@@ -52,25 +52,43 @@ void NumpyReaderGPU::Prefetch() {
         "The data produced by the reader has inconsistent dimensionality:\n"
         "[", data_idx, "] has ", sample->shape.size(), " dimensions whereas\n"
         "[0] has ", ref_shape.size(), " dimensions."));
-    DALI_ENFORCE(ref_shape.size() == sample->shape.size(),
-                  make_string("The tensors in the input batch do not all have the same number of "
-                  "dimensions for sample [0]: ", ref_shape.size(), " vs [", data_idx, "]: ",
-                  sample->shape.size(), "."));
     tmp_shapes.push_back(sample->shape);
   }
 
   curr_tensor_list.Resize(TensorListShape<>(tmp_shapes), ref_type);
 
+  size_t chunk_size = static_cast<size_t>( \
+                        div_ceil(static_cast<uint64_t>(curr_tensor_list.nbytes()), \
+                                 static_cast<uint64_t>(thread_pool_.NumThreads())));
+
   // read the data
   for (size_t data_idx = 0; data_idx < curr_tensor_list.ntensor(); ++data_idx) {
-    thread_pool_.AddWork([&curr_batch, &curr_tensor_list, data_idx](int tid) {
-        curr_batch[data_idx]->read_sample_f(curr_tensor_list.raw_mutable_data(),
-                                            curr_tensor_list.tensor_offset(data_idx) *
-                                            curr_tensor_list.type().size(),
-                                            curr_tensor_list.nbytes());
+    size_t image_bytes = static_cast<size_t>(volume(curr_tensor_list.tensor_shape(data_idx)) \
+                                             * curr_tensor_list.type().size());
```

Comment:
```suggestion
    size_t image_bytes = static_cast<size_t>(curr_tensor_list.tensor_size(data_idx) * curr_tensor_list.type().size());
```
nitpick: just a suggestion.